### ISSUE TITLE
Problem installing in MacOS

### ISSUE THREAD

**Issue by peremato** [HaarigerHarald/geant4_pybind#17]
I am trying to run some simple G4 examples to compare with https://github.com/JuliaHEP/Geant4.jl but I am having problems installing the extension on MacOS. 
- I tried with a X86 MacBook using the pre-built wheel but I get the following error after a successful `pip3 install geant4-pybind` :
```
>>> from geant4_pybind import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/geant4_pybind.cpython-39-darwin.so, 2): Symbol not found: __ZTTNSt3__114basic_ofstreamIcNS_11char_traitsIcEEEE
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/geant4_pybind.cpython-39-darwin.so
  Expected in: /usr/lib/libc++.1.dylib
 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/geant4_pybind.cpython-39-darwin.so
```
- I have also tried on a M1 MacMini, for which the wheel needs to be build, and I get the following error after having build correctly the dynamic library extension.
```
...
[490/490] Linking CXX shared module /private/var/folders/zz/gs4fh_9d5lx4k_q51gd7kvyr0000gn/T/pip-install-3_079jq7/geant4-pybind_d1557da9e313418685b29199fe58baf5/build/lib.macosx-13.4-arm64-cpython-39/geant4_pybind.cpython-39-darwin.so
Traceback (most recent call last):
  File "/private/var/folders/zz/gs4fh_9d5lx4k_q51gd7kvyr0000gn/T/pip-install-3_079jq7/geant4-pybind_d1557da9e313418685b29199fe58baf5/doc/generate_stubs.py", line 14, in <module>
    FunctionSignature.ignore_invalid_signature = True
NameError: name 'FunctionSignature' is not defined
Traceback (most recent call last):
  File "/Users/Shared/cvmfs/sft.cern.ch/lcg/views/LCG_104/arm64-mac13-clang140-opt/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
    main()
  File "/Users/Shared/cvmfs/sft.cern.ch/lcg/views/LCG_104/arm64-mac13-clang140-opt/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
...
```
Do you know how to fix these installation issues? Thanks in advance.

**Comment by HaarigerHarald** [comment]
There was an update in one of the tools used. I pushed a hotfix 3dac008 to use a previous version of that tool, which should allow you to build it.

The import error seems to be identical with #13. I can't say much about this as I'm not an expert on Apple products and I also don't have access to any Apple machines apart from the github CI runner (which seems to work fine).

**Comment by peremato** [comment]
Thanks. But I am not sure how to use the modified `pyproject.toml` with `pip3 install geant4-pybind`

**Comment by peremato** [comment]
It is fixed. I just had to clone with recursive submodules. Thanks again.

In case you are interested I run some very simple benchmarks.
- With B2a example with protons @ 3 GeV
- On my Mac-mini with the M1 processor

  | B2a (C++) | B2a.jl | B2a.py
-- | -- | -- | --
events = 1 | 0.9 s | 6 s | 1 s
events =100k | 106 s | 79 s | 176 s
events =100k (MT) | 23 s | 22 s | --

The reason why the python is slower is understandable because the user actions and sensitive detectors are implemented in Python and , it is many factors slower than C++ or Julia. If more work is done in these user actions, as it will be the case on more realistic applications, the performance will deteriorate  even further. I do to have an explanation why Julia should be faster than C++ in the serial case.  I was expecting within a few percent the same.

**Comment by HaarigerHarald** [comment]
The OSX import error should be fixed too with b89bb3afd11672f903f0e59e3db9557bdd023116 by building for an older OSX version.
